components/mpas-ocean/src/shared/mpas_ocn_tendency.F

```diff
@@ -235,6 +241,133 @@ subroutine ocn_tend_thick(tendPool, forcingPool)!{{{
+     normalizedRelativeVorticityEdge, normalizedPlanetaryVorticityEdge, &
+     montgomeryPotential, divergence, vertViscTopOfEdge, &
+     inSituThermalExpansionCoeff, inSituSalineContractionCoeff, vertVelocityNonhydro, &
+     tend_vertVelocityNonhydro
+
+   ! Scratch Arrays
+   ! normalThicknessFlux: Flux of thickness through an edge
+   !               units: m^{2} s^{-1}
+   real (kind=RKIND), dimension(:,:), allocatable :: normalThicknessFlux
+
+   integer :: timeLevel
+
+   integer :: err, iEdge, iCell, k
+   integer, pointer :: nVertLevels, indexTemperature, indexSalinity, nEdges, nCells
+   integer, dimension(:), pointer :: maxLevelCell
+
+   if (present(timeLevelIn)) then
+      timeLevel = timeLevelIn
+   else
+      timeLevel = 1
+   end if
+
+   call mpas_pool_get_dimension(meshPool, 'nEdges', nEdges)
+   call mpas_pool_get_dimension(meshPool, 'nCells', nCells)
+
+   call mpas_pool_get_dimension(meshPool, 'nVertLevels', nVertLevels)
+
+   call mpas_pool_get_array(statePool, 'vertVelocityNonhydro', vertVelocityNonhydro, timeLevel)
+   call mpas_pool_get_array(statePool, 'normalVelocity', normalVelocity, timeLevel)
+   call mpas_pool_get_array(statePool, 'layerThickness', layerThickness, timeLevel)
+   call mpas_pool_get_array(statePool, 'nonhydrostaticPressure', nonhydrostaticPressure, timeLevel)
+
+   call mpas_pool_get_array(tendPool, 'vertVelocityNonhydroTend', tend_vertVelocityNonhydro)
+
+   allocate(normalThicknessFlux(nVertLevels, nEdges+1))
+
+   !
+   ! transport velocity for the tracer.
+   !
+   !$omp parallel
+   !$omp do schedule(runtime) & 
+   !$omp private(k)
+   do iEdge = 1, nEdges
+      do k = 1, nVertLevels
+         normalThicknessFlux(k, iEdge) = normalTransportVelocity(k, iEdge) * layerThickEdgeFlux(k, iEdge)
+      end do
+   end do
+   !$omp end do
+   !$omp end parallel
+
+   !
+   ! velocity tendency: start accumulating tendency terms
+   !
+   !$omp parallel
+   !$omp do schedule(runtime)
+   do iCell = 1, nCells
+      tend_vertVelocityNonhydro(:, iCell) = 0.0_RKIND
+   end do
+   !$omp end do
+   !$omp end parallel
+
+   if(config_disable_vvel_all_tend) return
+
+   call mpas_timer_start("ocn_tend_vvel")
+   !
+   ! velocity tendency: nonlinear Coriolis term and grad of kinetic energy
+   !
+   call ocn_vvel_coriolis_tend(meshPool, normalVelocity, tend_vertVelocityNonhydro, err)
+
+   !
+   ! velocity tendency: horizontal and vertical advection
+   !
+   call ocn_vvel_advection_tend(meshPool, vertVelocityNonhydro, normalThicknessFlux, &
+                                        layerThickness, vertAleTransportTop, dt, &
+                                        tend_vertVelocityNonhydro)
+
+   !
+   ! velocity tendency: pressure gradient
+   !
+   call ocn_vvel_pgf_tend(meshPool, layerThickness, nonhydrostaticPressure, &
+        tend_vertVelocityNonhydro, err)
+
+   !
+   ! velocity tendency: del2 dissipation, CALL STRAIGHT to hmix_del2 FOR NOW!!!
```

Comment:
do we want to change this comment?  The call to hmix_del2 directly is quite different than tracer and normal velocity.  It may be fine to just make the comment read something like 'at present only del2 mixing is implemented for- horizontal mixing of vertical velocity'